src/dbt_client/dbtCommandFactory.ts

```diff
@@ -27,10 +27,24 @@ export class DBTCommandFactory {
     return dbtProfilesDir ? ["'--profiles-dir'", `r'${dbtProfilesDir.fsPath}'`] : [];
   }
 
+  private getFirstWorkspacePath(): string {
```

Comment:
I think we can also implement this differently. We can first detect any dbt projects and then launch the dbt detection on the first found dbt project.

This is related to simplifying the dbtProjectContainer vs dbtProject dependency wiring. We can get rid of a lot redirection functions by reducing the responsibilities of dbtProjectContainer.

Reply:
Because this is working as-is I suggest we move this to a separate PR encapsulating other dbtProjectContainer vs dbtProject changes where we can improve it and we get this one over the line.